### PR TITLE
Add logos to viewer

### LINKED ISSUE #3: Add feature - add engine logos
Graham has a set of engine logos.
Would be really cool if the logo of an engine would be shown next to its name in the web version of the broadcast

### PR DESCRIPTION
**Pull request by ianagbip1oti**
Fixes #3 

~~Adds a /logos/{engine} endpoint that searches for logos in a logos folder. If the filename is a prefix of the engine name, it will use that logo.~~

Sets the logo as a right aligned background image on the engine name.

The screenshots below show logos **at their worst**. It shows what happens if there is no logo. It shows what happens with long engine names that end up overlapping the logo.

Screenshot 2023-07-11 at 15 29 19
Screenshot 2023-07-11 at 15 29 30

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
+# Optional eslint cache
+.eslintcache
+
+# Misc
+.DS_Store
+
+.env
+*.pgn
+
+logos/
```
[the remark this answers is not in the document]
Comment (ianagbip1oti) [reply .gitignore:34]: I was thinking Graham keeps a zip, so we'd just want to dump that, but I'm not sure how you deploy stuff. We can commit them if that's preferred. I think there's a fair bit of tidying up of names and such to do though

File: src/logos.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+export function getLogo(engine: string): Logo | null {
+  const logos = fs.readdirSync(logos_dir);
+
+  logger.debug(`Searching ${logos.length} logos for ${engine}`);
+
+  const logo = logos.find((logo: string) => {
+    return engine.toLowerCase().startsWith(path.basename(logo).toLowerCase().split('.')[0]);
+  });
```
[the remark this answers is not in the document]
Comment (ianagbip1oti) [reply src/logos.ts:25]: We'd still want to do the search, as we have to deal with versions. We want the filename "princhess.bmp" to be found for `logo/princhess 0.11.0`. We can't really do a split, as the number of words in an engine name/version is too variable I believe.

[the remark this answers is not in the document]
Comment (jhonnold) [reply src/logos.ts:25]: Lowercase with substring where we stop at the first non [a-z]?

[the remark this answers is not in the document]
Comment (ianagbip1oti) [reply src/logos.ts:25]: I was referring more to the serving of static content. If there's an algorithm from name to logo file, then either it's done when the logo is requested (so on a not static asset endpoint), or it's done in the handlers somewhere and sent to the front end.

On the lowercase, I just dumped an archive of Graham's logos for testing, and they all have mixed case, hence I think we should support it. But, idk if you're seeing a world where we use those logos, or one where engine authors would make a PR with their logo.

[the remark this answers is not in the document]
Comment (ianagbip1oti) [reply src/logos.ts:25]: Ah, I see with an algorithm to figure out the name part of the engine without needing to know the files you could do that client side too

[the remark this answers is not in the document]
Comment (jhonnold) [reply src/logos.ts:25]: I will be converting them all to `webp` :)

[the remark this answers is not in the document]
Comment (ianagbip1oti) [reply src/logos.ts]: I've updated the PR based upon this discussion.

* Logos served from static logos folder (Princhess the only one added for now)
* Is a client side algorithm for trying to extract just the engine name from the engine string used for display
* Assumed to be webp

**Comment by jhonnold** [comment]
Graham is working on Logos and i'll do some additional verification with that once it's ready. Will probably merge and launch sometime this weekend.

**Comment by ianagbip1oti** [comment]
Awesome. Just added a small tweak where the engine may have digits in their engine name. e.g., CT800

**Review by jhonnold — APPROVED** [review]

**Pull request merged**